t/t-locks.sh

```diff
@@ -76,6 +76,34 @@ begin_test "list a single lock (SSH)"
 )
 end_test
 
+begin_test "list a single lock (SSH; git-lfs-transfer)"
+(
+  set -e
+
+  reponame="locks-list-ssh-pure"
+  setup_pure_ssh
+  setup_remote_repo_with_file "$reponame" "f.dat"
+  clone_repo "$reponame" "$reponame"
+
+  sshurl=$(ssh_remote "$reponame")
+  echo $sshurl
+  git config lfs.url "$sshurl"
+
+  GIT_TRACE_PACKET=1 git lfs lock --json "f.dat" | tee lock.log
+
+  id=$(assert_lock lock.log f.dat)
+  assert_server_lock_ssh "$reponame" "$id" "refs/heads/main"
+
+  GIT_TRACE=1 git lfs locks --path "f.dat" 2>trace.log | tee locks.log
+  cat trace.log
+  [ $(wc -l < locks.log) -eq 1 ]
+  grep "f.dat" locks.log
+  grep "lfs-ssh-echo.*git-lfs-transfer .*$reponame.git download" trace.log
+
+  git lfs unlock --id "$id"
+  refute_server_lock_ssh "$reponame" "$id" "refs/heads/main"
```

Comment:
The other tests in `t/t-locks.sh` don't seem to exercise the unlock functionality, which makes this new test a little special in that it's testing unlocking over SSH too, as well as it's stated purpose of `"list a single lock (SSH; git-lfs-transfer)"`.

Should tests for the the basic lock and unlock functionality over SSH be implemented in `t/t-lock.sh` and `t/t-unlock.sh`, respectively, leaving this new test in `t/t-locks.sh` just for checking the `git lfs locks` listing functionality?  (Obviously the test _also_ implicitly has to set up a lock first, which means `git lfs lock` will be exercised too, but that's more of a side-effect of setting up the test conditions than a properly separate test of the lock-taking functionality.)

Reply:
Yeah, I think we can do that.